completions/bfs.zsh

```diff
@@ -0,0 +1,176 @@
+#compdef bfs
+# Based on standard zsh find completion and bfs bash completion.
+
+local curcontext="$curcontext" state_descr variant default ret=1
+local -a state line args alts disp smatch
+
+args=(
+    # Flags
+    '(-depth)-d[search in post-order (descendents first)]'
+    '-D[print diagnostics]:debug option:(cost exec opt rates search stat time tree all help)'
+    '-E[use extended regular expressions with -regex/-iregex]'
+    '-f[specify file hierarchy to traverse]:path:_directories'
+    '-O+[enable query optimisation]:level:(1 2 3)'
+    '-s[traverse directories in sorted order]'
+    '-X[warn if filename contains characters special to xargs]'
+    "-x[don't span filesystems]"
+    '(-H -L)-P[never follow symlinks]'
+    '(-H -P)-L[follow symlinks]'
+    '(-L -P)-H[only follow symlinks when resolving command-line arguments]'
+    "-S[select search method]:value:(bfs dfs ids eds)"
+    '-f[treat path as path to search]:path:_files -/'
+
+    # Operators
+    '*-and'
+    '*-not'
+    '*-or'
+    '*-a' '*-o'
+    # TODO: Do we need to handle any of: , ! ( )
+    #       As they are single letters, can we even do anything useful?
```

Comment:
Not sure if that is possible when using the _arguments helper, it might require rewriting everything using lower level approaches:

```
_arguments:comparguments:325: invalid rest argument definition: *,
_arguments:comparguments:325: invalid rest argument definition: *!
_arguments:comparguments:325: invalid rest argument definition: *\(
```

I was able to get , and ! to work using another (quite hackish) approach, but I'm having trouble with ( and ). Even supporting , and ! is acting funny in some ways I don't quite understand. I have my zsh set up to get ls colors when completing files and directories, but that goes away if , and ! are in the completion list.

Reply:
I figured out a way to do it, and learnt more about the in depth workings of zsh completion while at it.